src/system/PlatformLockSupport.h

```diff
@@ -0,0 +1,34 @@
+/*
+ *    Copyright (c) 2025 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+/**
+ * The declared functions in this header are defined in src/platform/PlatformLockSupport.cpp because of dependency cycle issues.
```

Comment:
This is not and layering issue has to be fixed.

This header does not have any dependencies that I can see, so why is it not in platform then?

We need to figure out what depends on what else properly.